GpsDataAnalyzer/test/test_fileparser.py

```diff
@@ -60,18 +100,44 @@ def test_get_file_type(self):
         self.assertIsNone(none_file_type)
         
     def test_parse_xml_file_success(self):
-        xml_gpsdataset = self.fileparser.parse_file('testfile2.xml')
-        self.assertEqual(5, len(xml_gpsdataset.gpsdatalist))
+        xml_gpsdataset = self.fileparser.parse_file('test/testfile2.xml')
+        self.assertEqual(5, len(xml_gpsdataset.gps_data_list))
         self.assertEqual(self.xml_gpsdataset, xml_gpsdataset)
 
     def test_parse_invalid_file_return_none(self):
         none_gpsdataset = self.fileparser.parse_file('wrongfilename.')
         self.assertIsNone(none_gpsdataset)
 
-    def test_parse_csv_file_success(self):
-        #Todo
-        pass
-    
+    @patch('fileparser.utils.calculate_distance')
+    @patch('fileparser.utils.cartesian_to_geodetic')
+    def test_parse_csv_file_multiple_simulations(self, mock_cartesian_to_geodetic, mock_calculate_distance):
+        test_locations = [(37.31013773, -122.0314044, 44.6412353515625),
+                          (37.31013774, -122.03140539, 44.76190185546875),
+                          (37.31013773, -122.0314044, 44.6412353515625),
+                          (37.31013773, -122.0314044, 44.6412353515625)]
+        test_distances = [.016,]
+        mock_cartesian_to_geodetic.side_effect = test_locations
+        mock_calculate_distance.side_effect = test_distances
+        expected = [self.csv_gpsdataset1, self.csv_gpsdataset2, self.csv_gpsdataset3]
+
+        csv_data_set = self.fileparser.parse_csv('test/test_csv.csv')
+
+        self.assertEqual(csv_data_set, expected)
+
+    @patch('fileparser.utils.calculate_distance')
+    @patch('fileparser.utils.cartesian_to_geodetic')
+    def test_parse_csv_file_one_simulation(self, mock_cartesian_to_geodetic, mock_calculate_distance):
+        test_locations = [(37.31013773, -122.0314044, 44.6412353515625),
+                          (37.31013774, -122.03140539, 44.76190185546875)]
+        test_distances = [.016,]
+        mock_cartesian_to_geodetic.side_effect = test_locations
+        mock_calculate_distance.side_effect = test_distances
+        expected = [self.csv_gpsdataset1,]
+
+        csv_data_set = self.fileparser.parse_csv('test/test_csv_one_sim.csv')
+
+        self.assertEqual(csv_data_set, expected)
```

Comment:
Just curious, is it a convention for Python to keep objects order in (actual, expected)? Java's convention is (expected, actual).